src/nestedArray/types.ts

```diff
@@ -1,6 +1,11 @@
 import { DtypeString } from '../types';
 import { ValueError } from '../errors';
 
+declare global {
+  // eslint-disable-next-line
+  var Float16Array: import('@petamoriken/float16').Float16ArrayConstructor;
```

Comment:
I believe you can do `import type {Float16ArrayConstructor} from '@petamoriken/float16'` to only import the type and avoid something ending up in the bundle. I may be wrong but perhaps worth a try!

Reply:
The current babel config for jest does not support `type` imports like this, so this caused an issue. The inline type import `import('@petamoriken/float16').Float16ArrayConstructor;` is type-only and doesn't add anything to the bundle.

Reply:
This is no longer necessary, since we conditionally infer the type of `Float16Array` from globalThis.